splitio/engine/unique_keys_tracker.py

```diff
@@ -0,0 +1,95 @@
+import abc
+import threading
+import logging
+from splitio.engine.filters.bloom_filter import BloomFilter
+from splitio.engine.sender_adapters.in_memory_sender_adapter import InMemorySenderAdapter
+
+_LOGGER = logging.getLogger(__name__)
+
+class BaseUniqueKeysTracker(object, metaclass=abc.ABCMeta):
+    """Unique Keys Tracker interface."""
+
+    @abc.abstractmethod
+    def track(self, key, feature_name):
+        """
+        Return a boolean flag
+
+        """
+        pass
+
+class UniqueKeysTracker(BaseUniqueKeysTracker):
+    """Unique Keys Tracker class."""
+
+    def __init__(self, cache_size=30000):
+        """
+        Initialize unique keys tracker instance
+
+        :param cache_size: The size of the unique keys dictionary
+        :type key: int
+        """
+        self._cache_size = cache_size
+        self._filter = BloomFilter(cache_size)
+        self._lock = threading.RLock()
+        self._cache = {}
+        self._queue_full_hook = None
+
+    def track(self, key, feature_name):
+        """
+        Return a boolean flag
+
+        :param key: key to be added to MTK list
+        :type key: int
+        :param feature_name: split name associated with the key
+        :type feature_name: str
+
+        :return: True if successful
+        :rtype: boolean
+        """
+        if self._filter.contains(feature_name+key):
```

Comment:
this should be done within the lock context

Reply:
will do, I thought since it is only reading, no need for lock.